app/BUILD.bazel

```diff
@@ -258,16 +349,55 @@ in order to keep Gradle building.
  )
 for file in ALTERED_VIEWS]
 
+# Binding Adapters
 
 # TODO(#1617): Remove genrules post-gradle
-# Adapter files altered by genrules to be excluded in the app library.
+'''
+Binding adapter files under the org/oppia/app/databinding folder that need to be altered by the
+databinding binding adapter genrule.
+
+Please note that binding adapters have been split up into two types based on their location:
+databinding_adapters can be found under org/oppia/app/databinding and recyclerview_adapters can be
+found under org/oppia/app/recyclerview. Note that a genrule has not been created for
+recyclerview_adapters as it is not currently needed.
+
+Place your file here if:
+ - It is a binding adapter
+ - It imports resources (ex: `import org.oppia.app.R`)
+ - It is filed under org/oppia/app/databinding
+
+DO NOT place your file here if:
+ - It depends on layout files
+ - It does not import any resources
+ - It is found under org/oppia/app/recyclerview
+ - It uses Kotlin synthetic imports
+'''
 ALTERED_DATABINDING_ADAPTERS = [
```

Comment:
Given the requirements, maybe we should rename these to be DATABINDING_ADAPTERS_WITH_RESOURCE_IMPORTS and DATABINDING_ADAPTERS_WITHOUT_RESOURCE_IMPORTS to make things clearer. WDYT?

Ditto for the other 'altered' file lists.

Reply:
I think changing `ALTERED_*` to `*_WITH_RESOURCE_IMPORTS` makes sense. But I am not so sure about adding `_WITHOUT_RESOURCE_IMPORTS` - wouldn't this be confusing? The files processed with the genrule are added to this list and they still have resource imputs. This also ends up being the list included in the `src` attribute.

Reply:
I think what you added is clear. Thanks!